src/docs/cookbook/animation/physics-simulation.md

```diff
@@ -0,0 +1,393 @@
+---
+title: Animate a widget using a physics simulation
+prev:
+  title: Animate a page route transition
+  path: /docs/cookbook/animation/page-route-animation
+next:
+  title: Animate the properties of a container
+  path: /docs/cookbook/animation/animated-container
+---
+
+Physics simulations can make app interactions feel realistic and interactive.
+For example, you might want to animate a widget to act as if it were attached to
+a spring or falling with gravity.
+
+This recipe demonstrates how to move a widget from a dragged point back to the
+center using a spring simulation.
+
+This recipe uses these steps:
+
+1. Set up an animation controller
+2. Move the widget using gestures
+3. Animate the widget
+4. Calculate the velocity to simulate a springing motion
+
+
+## Step 1: Set up an animation controller
+
+Start with a stateful widget called `DraggableCard`:
+
+```dart
+import 'package:flutter/material.dart';
+
+main() {
+  runApp(MaterialApp(home: PhysicsCardDragDemo()));
+}
+
+class PhysicsCardDragDemo extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(),
+      body: DraggableCard(
+        child: FlutterLogo(
+          size: 128,
+        ),
+      ),
+    );
+  }
+}
+
+class DraggableCard extends StatefulWidget {
+  final Widget child;
+  DraggableCard({this.child});
+
+  @override
+  _DraggableCardState createState() => _DraggableCardState();
+}
+
+class _DraggableCardState extends State<DraggableCard> {
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Align(
+      child: Card(
+        child: widget.child,
+      ),
+    );
+  }
+}
+```
+
+Make the `_DraggableCardState` class extend from
+`SingleTickerProviderStateMixin`.  Then construct an `AnimationController` in
+`initState` and set `vsync` to `this`.
+
+{{site.alert.note}}
+Extending `SingleTickerProviderStateMixin` allows the state object to be a
```

Comment:
For future reference. The convention is to indent the note content by 2 spaces per line.